#!/usr/bin/env python
PACKAGE="micvision"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("inflation_radius", double_t, 0, "The inflation radius(m)", 0.3, 0, 1)
gen.add("robot_radius", double_t, 0, "The robot radius(m)", 0.2, 0, 0.5)
gen.add("laserscan_circle_step", int_t, 0, "Laser scan sampled every circle", 1, 1, 10)
gen.add("range_step", int_t, 0, "Robot location sample step(pixel)", 3, 1, 10)
gen.add("laserscan_anglar_step", int_t, 0, "At each location, we sampled the anglar(degree)", 6, 1, 10)

gen.add("min_valid_range", double_t, 0, "The minimum valid range(m)", 0.0, 0.0, 1)
gen.add("max_valid_range", double_t, 0, "The maximum valid range(m)", 10.0, 0.0, 40.0)

gen.add("quick_score", bool_t, 0, "Using quick score", True)
gen.add("quick_score_num", int_t, 0, "The quick score number", 8, 4, 20)

exit(gen.generate(PACKAGE, "micvision", "Location"))
